receiver/scraperhelper/scrapercontroller.go

```diff
@@ -182,14 +167,19 @@ func (sc *controller) startScraping() {
 			sc.tickerCh = ticker.C
 		}
 
+		ctx, done := withScrapeContext(sc.timeout)
+
 		// Call scrape method on initialision to ensure
 		// that scrapers start from when the component starts
 		// instead of waiting for the full duration to start.
-		sc.scrapeMetricsAndReport(context.Background())
+		sc.scrapeMetricsAndReport(ctx)
+		done()
```

Comment:
Since we're doing the exact same thing here and within the loop, would it make sense to move the context work into `scrapeMetricsAndReport`? This would allow us to `defer done()`, which I think it more typical.

Reply:
... I should have noticed this